plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -100,36 +100,25 @@ protected SR createFileSR(final Connection conn, final String path) {
         PBD pbd = null;
 
         try {
-            final String srname = hypervisorResource.getHost().getUuid() + path.trim();
-
-            final Set<SR> srs = SR.getByNameLabel(conn, srname);
-
-            if (srs != null && !srs.isEmpty()) {
-                return srs.iterator().next();
+            final String srname = path.trim();
+            synchronized (srname.intern()) {
```

Comment:
Should we consider using a `ReentrantLock` instead of `synchronized`?  `ReentrantLock.tryLock(long, TimeUnit)` allows the specification of a timeout to acquire the lock.  Since a timeout cannot be specified to `synchronized`, multiple threads may deadlock attempting to acquire the lock in the case of a stall.  The downside of this approach is that we would need to track locks in a `ConcurrentMap` -- ideally with WeakReferences to allow them to be garbage collected when no threads are operating on them.